app/(root)/(home)/page.tsx

```diff
@@ -1,25 +1,61 @@
 import { getUser } from "@/lib/actions/user.actions";
+import {
+  findPosts,
+  getAllPosts,
+  getFilteredPosts,
+} from "@/lib/actions/post.actions";
 
-import { User } from "@prisma/client";
-// import Posts from "../posts/page";
+import { CreateType, Post, User } from "@prisma/client";
+import Posts from "../../../components/post/Posts";
+import Pagination from "@/components/post/Pagination";
 
-export default async function Home() {
+export default async function Home({
+  searchParams,
+}: {
+  searchParams: { page: string; type: string };
+}) {
   const user = (await getUser()) as User;
 
+  const posts = await getAllPosts({
+    page: searchParams.page ?? "1",
+  });
+  const { somePosts, hasNextPage, numberOfPages } = posts;
+
+  const filteredPosts = await getFilteredPosts({
+    page: searchParams.page ?? "1",
+    searchTerm: searchParams.type as CreateType,
+  });
+  const { someFilteredPosts, hasNextPageFiltered, numberOfPagesFiltered } =
+    filteredPosts;
+
   return (
-    <>
-      <div className="text-white-300 flex flex-col gap-4">
-        <h1 className="display-2-bold text-white-100">
-          Hi {user && user.name}!
-        </h1>
-        <p className="paragraph-3-medium">
-          You have the role of {user && user.role}
-        </p>
-      </div>
-      <div className="text-white-300 mt-4 flex flex-col gap-4">
-        <h3 className="display-2-bold text-white-100">Recent Posts</h3>
-        {/* {session && <Posts user={user} />} */}
+    <section className="flex flex-col gap-4">
+      <h1 className="display-1-bold text-white-100">
+        Hello {user && user.name} {searchParams.page},
+      </h1>
+      <p className="paragraph-1-regular text-white-300">
+        Time to jot down your latest learnings today!
+      </p>
+      <div className="bg-black-700 h-52 w-full">
+        {/* placeholder for contribution grid */}
       </div>
-    </>
+      {somePosts && (searchParams.type === "all" || !searchParams.type) ? (
+        <>
+          <Posts posts={somePosts as Post[]} />
+          <Pagination
+            hasNextPage={hasNextPage as boolean}
+            numberOfPages={numberOfPages as number}
+          />
+        </>
+      ) : (
+        <>
+          <Posts posts={someFilteredPosts as Post[]} />
+          <Pagination
+            hasNextPage={hasNextPageFiltered as boolean}
+            numberOfPages={numberOfPagesFiltered as number}
+          />
+        </>
+      )}
```

Comment:
```suggestion
      {somePosts && (
        <>
          <Posts posts={somePosts as Post[]} />
          <Pagination
            hasNextPage={hasNextPage as boolean}
            numberOfPages={numberOfPages as number}
          />
        </>
      )}
```

Let's simplify this since we only have one query now.